extensions/sql/contract-definition/store/src/main/java/org/eclipse/dataspaceconnector/sql/contractdefinition/store/SqlContractDefinitionStore.java

```diff
@@ -0,0 +1,244 @@
+    }
+
+    @SuppressWarnings("unchecked")
+    private static <T> T readObject(String value) {
+        try {
+            Envelope envelope = OBJECT_MAPPER.readValue(value, Envelope.class);
+            return (T) OBJECT_MAPPER.readValue(envelope.getContent(), Class.forName(envelope.getClassName()));
+        } catch (Exception e) {
+            throw new EdcException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public @NotNull Collection<ContractDefinition> findAll() {
+        String query = "SELECT * from %s";
+
+        try (Connection connection = dataSource.getConnection()) {
+            return SqlQueryExecutor.executeQuery(
+                    connection,
+                    ContractDefinitionMapper.INSTANCE,
+                    String.format(query, SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE));
+        } catch (Exception exception) {
+            throw new EdcPersistenceException(exception);
+        }
+    }
+
+    @Override
+    public @NotNull Stream<ContractDefinition> findAll(QuerySpec spec) {
+        Objects.requireNonNull(spec);
+
+        Limit limit = Limit.Builder.newInstance()
+                .limit(spec.getLimit())
+                .offset(spec.getOffset())
+                .build();
+
+        String query = "SELECT * from %s " + limit.getStatement();
+
+        try (Connection connection = dataSource.getConnection()) {
+            List<ContractDefinition> definitions = SqlQueryExecutor.executeQuery(
+                    connection,
+                    ContractDefinitionMapper.INSTANCE,
+                    String.format(query, SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE));
+            return definitions.stream();
+        } catch (Exception exception) {
+            throw new EdcPersistenceException(exception);
+        }
+    }
+
+    @Override
+    public void save(Collection<ContractDefinition> definitions) {
+        Objects.requireNonNull(definitions);
+
+        String query = String.format("INSERT INTO %s (%s, %s, %s, %s) VALUES (?, ?, ?, ?)",
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_ID,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_ACCESS_POLICY,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_CONTRACT_POLICY,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_SELECTOR);
+
+        transactionContext.execute(() -> {
+            try (Connection connection = dataSource.getConnection()) {
+                definitions.forEach((definition) -> executeQuery(connection, query,
+                        definition.getId(),
+                        writeObject(definition.getAccessPolicy()),
+                        writeObject(definition.getContractPolicy()),
+                        writeObject(definition.getSelectorExpression())));
+            } catch (Exception exception) {
```

Comment:
Unneccessary, because the `SqlQueryExecutor` already wraps every `Exception` in an `EdcPersistenceException`